Detectors/HMPID/simulation/src/Detector.cxx

```diff
@@ -429,12 +515,532 @@ TGeoVolume* Detector::createChamber(int number)
+
 void Detector::ConstructGeometry()
 {
+  // Creates detailed geometry simulation (currently GEANT volumes tree)
+  // includind the HMPID cradle
+
   createMaterials();
-  // for the moment create just 1 chamber and connect to the cave volume
-  auto vol = createChamber(0);
-  gGeoManager->GetVolume("cave")->AddNode(vol, 0);
+
+  TGeoVolume* hmpcradle = CreateCradle();
+
+  for (Int_t iCh = 0; iCh <= 6; iCh++) { // place 7 chambers
+    TGeoVolume* hmpid = createChamber(iCh);
+    TGeoHMatrix* pMatrix = new TGeoHMatrix;
+    IdealPosition(iCh, pMatrix);
+    gGeoManager->GetVolume("cave")->AddNode(hmpid, 0, pMatrix);
+    if (iCh == 1 || iCh == 3 || iCh == 5) {
+      TGeoHMatrix* pCradleMatrix = new TGeoHMatrix;
+      IdealPositionCradle(iCh, pCradleMatrix);
+      gGeoManager->GetVolume("cave")->AddNode(hmpcradle, iCh, pCradleMatrix);
+    }
+  }
+  // }
+  // AliDebug(1,"Stop v3. HMPID option");
+}
+
+//*****************************************************************************************************************
+void Detector::DefineOpticalProperties()
+{
+  // AliDebug(1,"");
+
+  // Optical properties definition.
+  const Int_t kNbins = 30;        // number of photon energy points
+  Float_t emin = 5.5, emax = 8.5; // Photon energy range,[eV]
+  Float_t deltaE = (emax - emin) / kNbins;
+  Float_t aEckov[kNbins];
+  Double_t dEckov[kNbins];
+  Float_t aAbsRad[kNbins], aAbsWin[kNbins], aAbsGap[kNbins], aAbsMet[kNbins];
+  Float_t aIdxRad[kNbins], aIdxWin[kNbins], aIdxGap[kNbins], aIdxMet[kNbins], aIdxPc[kNbins];
+  Float_t aQeAll[kNbins], aQePc[kNbins];
+  Double_t dReflMet[kNbins], dQePc[kNbins];
+
+  TF2* pRaIF = new TF2("HidxRad", "sqrt(1+0.554*(1239.84/x)^2/((1239.84/x)^2-5769)-0.0005*(y-20))", emin, emax, 0,
+                       50); // DiMauro mail temp 0-50 degrees C
+  TF1* pWiIF = new TF1("HidxWin", "sqrt(1+46.411/(10.666*10.666-x*x)+228.71/(18.125*18.125-x*x))", emin,
+                       emax);                                                                  // SiO2 idx TDR p.35
+  TF1* pGaIF = new TF1("HidxGap", "1+0.12489e-6/(2.62e-4 - x*x/1239.84/1239.84)", emin, emax); //?????? from where
+
+  TF1* pRaAF =
+    new TF1("HabsRad", "(x<7.8)*(gaus+gaus(3))+(x>=7.8)*0.0001", emin, emax); // fit from DiMauro data 28.10.03
+  pRaAF->SetParameters(3.20491e16, -0.00917890, 0.742402, 3035.37, 4.81171, 0.626309);
+  TF1* pWiAF = new TF1("HabsWin", "(x<8.2)*(818.8638-301.0436*x+36.89642*x*x-1.507555*x*x*x)+(x>=8.2)*0.0001", emin,
+                       emax); // fit from DiMauro data 28.10.03
+  TF1* pGaAF = new TF1(
+    "HabsGap", "(x<7.75)*6512.399+(x>=7.75)*3.90743e-2/(-1.655279e-1+6.307392e-2*x-8.011441e-3*x*x+3.392126e-4*x*x*x)",
+    emin, emax); //????? from where
+
+  TF1* pQeF = new TF1("Hqe", "0+(x>6.07267)*0.344811*(1-exp(-1.29730*(x-6.07267)))", emin,
+                      emax); // fit from DiMauro data 28.10.03
```

Comment:
do these histograms need to  be pointers? Prefer using straight objects or unique pointers to avoid having to delete these things manually.